scripts/wikipedia.js

```diff
@@ -0,0 +1,66 @@
+// This script adds buttons next to isbns on wikipedia pages that will redirect
+// the user to a readable digital copy of the referenced book.
+
+
+//Get all books on wikipedia page through https://archive.org/services/context/books?url=...
+function getJSON(){
+  let url = "https://archive.org/services/context/books?url=" + location.href;
+  fetch(url)
+    .then(res => res.json())
+    .then(data => {
+      let books = document.getElementsByClassName('citation book');
+      for(let book of books){
+        let isbn = getISBNFromCitation(book);
+        let id = getIdentifierFromISBN(isbn, data);
+        if(id){
+          let link = createLinkToArchive(id);
+          book.appendChild(link);
+        }
+      }
+    })
+    .catch(err=> console.log(err));
+}
```

Comment:
In https://github.com/internetarchive/wayback-machine-chrome/blob/master/scripts/booklist.js we have similar functionality, we access the /books endpoint and get data to do something else there.

Can we make a function like `get_ia_books(url)` in `scripts/booklist.js` and use it both there and also here in this function? Its not good to have duplicate code.
Thank you!

Reply:
Good idea.  I am making the `get_ia_books(url)` function in `scripts/wikipedia.js` instead, and it returns a promise.